### PR TITLE
feat: roundtrip fuzz harness for `PooledTransactions`

### PR DESCRIPTION
**Pull request by 0xalpharush**
Add a roundtrip fuzz harness for `PooledTransactions` by converting an existing unit test into a fuzz test. This harness identified a new bug (cc @Rjected). Run the harness with `cargo test-fuzz -p reth-eth-wire --run-until-crash` and reproduce the crash with `cargo test-fuzz -p reth-eth-wire --replay crashes`.

Partially addresses https://github.com/paradigmxyz/reth/issues/4962

### CODE REVIEW

**Comment by mattsse** [comment]
what's the bug?

**Comment by Rjected** [comment]
> what's the bug?

https://github.com/recmo/uint/pull/335 is one surfaced by this fuzz test

Another is related to payload length checks, PR soon

**Review by Rjected — CHANGES_REQUESTED** [review]
Just one comment about preserving existing behavior from `decode_pooled_transactions_data` in at least one test

File: crates/net/eth-wire/tests/pooled_transactions.rs [hunk 1]
```diff
@@ -29,6 +31,15 @@ fn decode_pooled_transactions_data() {
...
 }
 
+#[test]
+fn decode_pooled_transactions_data() {
+    let network_data_path =
+        PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("testdata/pooled_transactions_with_blob");
+    let data = fs::read_to_string(network_data_path).expect("Unable to read file");
+    let hex_data = hex::decode(data.trim()).unwrap();
+    roundtrip_pooled_transactions(hex_data);
+}
```
[the remark this answers is not in the document]
Comment (0xalpharush) [reply crates/net/eth-wire/tests/pooled_transactions.rs]: I added an assertion that the existing tests decodes successfully in https://github.com/paradigmxyz/reth/pull/5125/commits/9eb594c8d0734b3cba383619b941c8a9fd6a6579

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected

**Review by Rjected — APPROVED** [review]
LGTM!

**Pull request merged**